azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java

```diff
@@ -399,8 +407,36 @@ protected WebAppConfig getWebAppConfig() throws AzureExecutionException {
         return parser.parse();
     }
 
-    protected AzureAppService getOrCreateAzureAppServiceClient() {
-        return Azure.az(AzureAppService.class);
+    protected AzureAppService getOrCreateAzureAppServiceClient() throws AzureExecutionException {
+        if (appServiceClient == null) {
+            try {
+                final Account account = getAzureAccount();
+                final List<Subscription> subscriptions = account.getSubscriptions();
+                final String targetSubscriptionId = getTargetSubscriptionId(getSubscriptionId(), subscriptions, account.getSelectedSubscriptions());
```

Comment:
I think we should move the selection of one target subscription in getAzureAccount by calling select subscription, then we will not need to have the logic like:

appServiceClient == null ? this.subscriptionId : appServiceClient.getDefaultSubscription().getId();

public String getSubscriptionId() {
        return mojo.getSubscriptionId();
    }

Reply:
Yes, for command line tooling, as there will always be one subscription, we may select it in Account and always use the selected one, this should be done in later PR for all command line tooling.